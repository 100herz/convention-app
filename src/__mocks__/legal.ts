import { Legal } from '@models/legal'

export const legalArray: Legal[] = [
  { id: 1, name: 'Impressum' },
  { id: 2, name: 'Datenschutz' },
]
