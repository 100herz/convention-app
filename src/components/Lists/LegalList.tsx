import React from 'react'
import { StyleSheet, View } from 'react-native'
import { Ionicons } from '@expo/vector-icons'
// import { useNavigation } from '@react-navigation/native'
// import { StackNavigationProp } from '@react-navigation/stack'

import Text from '@components/UI/Text'
import Touchable from '@components/UI/Touchable'
// import { RootStackParamList } from '@navigations/HomeNavigator'
import { Legal } from '@models/Legal'
import { colors, defaultStyles, DefaultStyles } from '@styles/theme'

interface Props {
  data: Legal[]
}

const LegalList: React.FC<Props> = ({ data }) => {
  // const navigation = useNavigation<StackNavigationProp<RootStackParamList>>()

  return (
    <View style={styles.listContainer}>
      {data.map(item => (
        <Touchable
          key={item.id}
          // onPress={() => navigation.navigate('ArticlesOverviewScreen', { LegalId: item.id, LegalName: item.name })}
          onPress={() => {}} // eslint-disable-line @typescript-eslint/no-empty-function
          testID="link"
        >
          <View style={styles.listItemContainer}>
            <Text>{item.name}</Text>
            <Ionicons name="ios-arrow-round-forward" size={20} color={colors.primaryColor} />
          </View>
        </Touchable>
      ))}
    </View>
  )
}

const styles = StyleSheet.create<DefaultStyles>({
  ...defaultStyles,
})

export default LegalList
