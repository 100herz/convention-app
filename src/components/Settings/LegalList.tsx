import React from 'react'
import { Linking, StyleSheet, View } from 'react-native'
import { Ionicons } from '@expo/vector-icons'
import { useNavigation } from '@react-navigation/native'
import { StackNavigationProp } from '@react-navigation/stack'
import HTML from 'react-native-render-html'

import Touchable from '@components/UI/Touchable'
import Text from '@components/UI/Text'
import { SettingsStackParamList } from '@navigations/SettingsNavigator'
import { Legal } from '@models/settings'
import { legalAddition } from '@data/settings'
import { colors, defaultStyles, DefaultStyles } from '@styles/theme'

interface Props {
  data: Legal[]
}

const LegalList: React.FC<Props> = ({ data }) => {
  const navigation = useNavigation<StackNavigationProp<SettingsStackParamList>>()

  const linkIcon = () => <Ionicons name="ios-arrow-round-forward" size={20} color={colors.primaryColor} />

  return (
    <View style={styles.listContainer}>
      {data.map(item => (
        <Touchable
          key={item.id}
          onPress={() => navigation.navigate('LegalScreen', { pageId: item.id, screenTitle: item.title.rendered })}
          testID={`link-${item.id}`}
        >
          <View style={styles.listItemContainer}>
            <HTML baseFontStyle={styles.text} html={item.title.rendered} />
            {linkIcon()}
          </View>
        </Touchable>
      ))}

      <Touchable onPress={() => Linking.openURL(legalAddition.href)} testID="link-mail">
        <View style={styles.listItemContainer}>
          <Text>{legalAddition.name}</Text>
          {linkIcon()}
        </View>
      </Touchable>
    </View>
  )
}

const styles = StyleSheet.create<DefaultStyles>({
  ...defaultStyles,
})

export default LegalList
