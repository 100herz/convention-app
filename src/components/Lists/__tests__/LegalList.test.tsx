import React from 'react'
import { render } from '@testing-library/react-native'

import LegalList from '../LegalList'
import { legalArray } from '@__mocks__/legal'

describe('<LegalList />', () => {
  it('renders correctly', () => {
    const { toJSON } = render(<LegalList data={legalArray} />)
    expect(toJSON()).toMatchSnapshot()
  })
})
