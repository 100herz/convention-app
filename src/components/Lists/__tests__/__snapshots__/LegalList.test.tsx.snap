// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LegalList /> renders correctly 1`] = `
<RCTScrollView
  data={
    Array [
      Object {
        "id": 1,
        "name": "Impressum",
      },
      Object {
        "id": 2,
        "name": "Datenschutz",
      },
    ]
  }
  disableVirtualization={false}
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={false}
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  onContentSizeChange={[Function]}
  onEndReachedThreshold={2}
  onLayout={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={Array []}
  style={
    Object {
      "paddingTop": 15,
    }
  }
  updateCellsBatchingPeriod={50}
  viewabilityConfigCallbackPairs={Array []}
  windowSize={21}
>
  <View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID="link"
      >
        <View
          style={
            Object {
              "borderBottomColor": "#dfe3e8",
              "borderBottomWidth": 1,
              "flexDirection": "row",
              "justifyContent": "space-between",
              "paddingHorizontal": 15,
              "paddingVertical": 10,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#333333",
                "fontFamily": "sans",
              }
            }
          >
            Impressum
          </Text>
          <Text />
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID="link"
      >
        <View
          style={
            Object {
              "borderBottomColor": "#dfe3e8",
              "borderBottomWidth": 1,
              "flexDirection": "row",
              "justifyContent": "space-between",
              "paddingHorizontal": 15,
              "paddingVertical": 10,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#333333",
                "fontFamily": "sans",
              }
            }
          >
            Datenschutz
          </Text>
          <Text />
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
